src/storage/src/hummock/file_cache/store.rs

```diff
@@ -1,803 +0,0 @@
-    pub fn new(inner: CachedBlock) -> Self {
-        Self { inner, pos: 0 }
-    }
-}
-
-impl std::io::Read for CachedBlockCursor {
-    fn read(&mut self, mut buf: &mut [u8]) -> std::io::Result<usize> {
-        let pos = self.pos;
-        match &self.inner {
-            CachedBlock::Loaded { block } => {
-                if self.pos < 1 {
-                    self.pos += copy([0], &mut buf);
-                }
-                self.pos += copy(&block.raw()[self.pos - 1..], &mut buf);
-            }
-            CachedBlock::Fetched {
-                bytes,
-                uncompressed_capacity,
-            } => {
-                if self.pos < 1 {
-                    self.pos += copy([1], &mut buf);
-                }
-                if self.pos < 9 {
-                    self.pos += copy(
-                        &uncompressed_capacity.to_be_bytes()[self.pos - 1..],
-                        &mut buf,
-                    );
-                }
-                self.pos += copy(&bytes[self.pos - 9..], &mut buf);
-            }
-        }
-        let n = self.pos - pos;
-        Ok(n)
-    }
-}
-
-impl Cursor for CachedBlockCursor {
-    type T = CachedBlock;
-
-    fn into_inner(self) -> Self::T {
-        self.inner
-    }
-}
-
-impl Value for Box<Block> {
-    type Cursor = BoxBlockCursor;
-
-    fn serialized_len(&self) -> usize {
-        self.raw().len()
-    }
-
-    fn read(buf: &[u8]) -> CodingResult<Self> {
-        let data = Bytes::copy_from_slice(buf);
-        let block = Block::decode_from_raw(data);
-        let block = Box::new(block);
-        Ok(block)
-    }
-
-    fn into_cursor(self) -> Self::Cursor {
-        BoxBlockCursor::new(self)
-    }
-}
-
-#[derive(Debug)]
-pub struct BoxBlockCursor {
-    inner: Box<Block>,
-    pos: usize,
-}
-
-impl BoxBlockCursor {
-    pub fn new(inner: Box<Block>) -> Self {
-        Self { inner, pos: 0 }
-    }
-}
-
-impl std::io::Read for BoxBlockCursor {
-    fn read(&mut self, buf: &mut [u8]) -> std::io::Result<usize> {
-        let pos = self.pos;
-        self.pos += copy(&self.inner.raw()[self.pos..], buf);
-        let n = self.pos - pos;
-        Ok(n)
-    }
-}
-
-impl Cursor for BoxBlockCursor {
-    type T = Box<Block>;
-
-    fn into_inner(self) -> Self::T {
-        self.inner
-    }
-}
-
-#[derive(Debug)]
-pub struct CachedSstable(Arc<Sstable>);
-
-impl Clone for CachedSstable {
-    fn clone(&self) -> Self {
-        Self(Arc::clone(&self.0))
-    }
-}
-
-impl From<Box<Sstable>> for CachedSstable {
-    fn from(value: Box<Sstable>) -> Self {
-        Self(Arc::new(*value))
-    }
-}
-
-impl From<CachedSstable> for Box<Sstable> {
-    fn from(value: CachedSstable) -> Self {
-        Box::new(Arc::unwrap_or_clone(value.0))
-    }
-}
-
-impl CachedSstable {
-    pub fn into_inner(self) -> Arc<Sstable> {
-        self.0
-    }
-}
-
-impl Value for CachedSstable {
-    type Cursor = CachedSstableCursor;
-
-    fn weight(&self) -> usize {
-        self.0.estimate_size()
```

Comment:
This is a question rather than a comment. Previously weight can be different from the serialized_len of the value. After this PR, we no longer use the value trait and use Serialize instead. Does it mean that we use the serialized len as the weight now?

Reply:
With the old implementation, the tiered cache uses `weight` for in-memory cache and `serialized_len` for disk cache.

With the new foyer, to make `Key` and `Value` as easy to use as possible, `Key` and `Value` doesn't require any function like `weight` anymore. The customized weighter interface is moved to the in-memory cache build, which receives a weighter function:

https://github.com/risingwavelabs/risingwave/pull/16500/files#diff-37c04fbadb030dfc3e5470535330f9cfa9e6a4e7ebf0ed2b238c799eddd11441R583-R585

And for the disk cache part,  it doesn't requires `serialized_len` anymore. For compression, the compressed len cannot be decided early. So the `serialized_len` was only used for metrics and hints for a while. And it is finally removed by the new version.